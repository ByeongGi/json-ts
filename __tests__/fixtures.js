const {json2ts} = require('../');
const {readFileSync} = require('fs');
const data = (name) => readFileSync(name, 'utf8');

it('matches gmaps', function () {
    const gmaps = data('__tests__/gmaps/in.json');
    expect(json2ts(gmaps)).toMatchSnapshot();
});

it('matches magento', function () {
    const gmaps = data('__tests__/magento/product.json');
    expect(json2ts(gmaps)).toMatchSnapshot();
});

it('matches petition', function () {
    const gmaps = data('__tests__/petition/input.json');
    expect(json2ts(gmaps)).toMatchSnapshot();
});