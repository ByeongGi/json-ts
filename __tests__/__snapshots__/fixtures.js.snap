// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches gmaps 1`] = `
"interface IRootObject {
    results: IResultsItem[];
    status: string;
}

interface IResultsItem {
    address_components: IAddressComponentsItem[];
    formatted_address: string;
    geometry: IGeometry;
    place_id: string;
    types: string[];
}

interface IAddressComponentsItem {
    long_name: string;
    short_name: string;
    types: string[];
}

interface IGeometry {
    bounds: IBounds;
    location: ILocation;
    location_type: string;
    viewport: IViewport;
}

interface IBounds {
    northeast: INortheast;
    southwest: ISouthwest;
}

interface INortheast {
    lat: number;
    lng: number;
}

interface ISouthwest {
    lat: number;
    lng: number;
}

interface ILocation {
    lat: number;
    lng: number;
}

interface IViewport {
    northeast: INortheast;
    southwest: ISouthwest;
}
"
`;

exports[`matches magento 1`] = `
"interface IRootObject {
    id: number;
    sku: string;
    name: string;
    attribute_set_id: number;
    price: number;
    status: number;
    visibility: number;
    type_id: string;
    created_at: string;
    updated_at: string;
    extension_attributes: any[];
    product_links: any[];
    options: any[];
    media_gallery_entries: IMediaGalleryEntriesItem[];
    tier_prices: any[];
    custom_attributes: ICustomAttributesItem[];
}

interface IMediaGalleryEntriesItem {
    id: number;
    media_type: string;
    label: string;
    position: number;
    disabled: boolean;
    types: string[];
    file: string;
}

interface ICustomAttributesItem {
    attribute_code: string;
    value: string | string[];
}
"
`;

exports[`matches magento categories 1`] = `
"interface IRootObject {
    id: number;
    parent_id: number;
    name: string;
    image: null;
    is_active: boolean;
    include_in_menu: boolean;
    position: number;
    level: number;
    product_count: number;
    children_data: IChildrenDataItem[];
}

interface IChildrenDataItem {
    id: number;
    parent_id: number;
    name: string;
    image: null | string;
    is_active: boolean;
    include_in_menu: boolean;
    position: number;
    level: number;
    product_count: number;
    children_data: IChildrenDataItem[];
}
"
`;

exports[`matches petition 1`] = `
"interface IRootObject {
    format: string;
    version: string;
    result: IResult;
}

interface IResult {
    _about: string;
    definition: string;
    extendedMetadataVersion: string;
    first: string;
    hasPart: string;
    isPartOf: string;
    items: IItemsItem[];
    itemsPerPage: number;
    next: string;
    page: number;
    startIndex: number;
    totalResults: number;
    type: string[];
}

interface IItemsItem {
    _about: string;
    created: ICreated;
    identifier: IIdentifier;
    label: ILabel;
    numberOfSignatures: number;
    status: string;
}

interface ICreated {
    _value: string;
    _datatype: string;
}

interface IIdentifier {
    _value: string;
}

interface ILabel {
    _value: string;
}
"
`;
